pkg/server/metrics.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2020 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"net/http"
+
+	"k8s.io/apiserver/pkg/server/mux"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+// DefaultMetrics installs the default prometheus metrics handler
+type DefaultMetrics struct {
+	registry metrics.Gatherer
+}
+
+// Install adds the DefaultMetrics handler
+func (m DefaultMetrics) Install(c *mux.PathRecorderMux) {
+	c.HandleFunc("/metrics", func(w http.ResponseWriter, req *http.Request) {
+		legacyregistry.Handler().ServeHTTP(w, req)
```

Comment:
what metrics legacyregistry provides? Could we remove it?

Reply:
The API server metrics are still registered to the global registry. AFAIK, there is no mechanism to dependency inject a registry for these metrics, so we still have to serve the default handler until a similar effort is done in the API server.